### PR TITLE
RavenDB-16206 Query builder optimization.

### PR DESCRIPTION
**Pull request by maciejaszyk**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-16206

### Additional description

In most cases, users use the standard "p" parameter so we are generating 128 first parameters at the startup.

### Type of change

- Optimization

### How risky is the change?

- Not relevant

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

-Covered by current tests.

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

**Review by ayende — COMMENTED** [review]
Can you add some profiling details? What was the overall impact?

**Comment by maciejaszyk** [comment]
> Can you add some profiling details? What was the overall impact?
image
Almost 3x faster in most cases.

File: src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs [hunk 1]
```diff
@@ -1743,12 +1744,16 @@ private object TransformValue(WhereParams whereParams, bool forRange = false)
...
+            if (ParameterPrefix == "p" && QueryParameters.Count < 128)
+                parameterName = _pregeneratedParametersForPrefixP[QueryParameters.Count];
+            else
+                parameterName = $"{ParameterPrefix}{QueryParameters.Count.ToInvariantString()}";
             QueryParameters.Add(parameterName, value);
             return parameterName;
         }
 
-        public string ParameterPrefix { get; set; } = "p";
+        public string ParameterPrefix { get; set; }  = "p";
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs]: introduce prvate const inside this class `DefaultParameterPrefix` and set that to `p` and use it both places.

Comment (maciejaszyk) [reply src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs]: done

**Pull request merged**